### PR TITLE
Implement device reset command

### PR DESCRIPTION
**Pull request by obbardc**
Add a command to allow the device to be reset.

### CODE REVIEW

File: rockusb/src/protocol.rs [hunk 1]
```diff
@@ -173,6 +183,7 @@ pub struct CommandBlock {
     cdb_length: u8,
     // Command data block fields
     cd_code: CommandCode,
+    cd_reserved: u8,
```
Comment (sjoerdsimons) [inline rockusb/src/protocol.rs:186]: if it's used it's clearly not reserved

File: rockusb/src/operation.rs [hunk 2]
```diff
@@ -244,7 +244,11 @@ where
         match next {
             Operation::CommandBlock => {
                 let len = self.command.to_bytes(&mut self.command_bytes);
+                // If there is no transfer to be made, skip the transfer state.
                 self.next = Operation::IO;
+                if self.command.transfer_length() == 0 {
```
Comment (sjoerdsimons) [inline rockusb/src/operation.rs:249]: if/else block please

File: rockusb/src/protocol.rs (hunk 1 above)
Comment (obbardc) [reply rockusb/src/protocol.rs:186]: it's an optional parameter to the command; in rkdeveloptool they call it a "subcode". What do you think about `cd_subcode` ?

File: rockusb/examples/rockusb.rs [hunk 3]
```diff
@@ -29,6 +30,12 @@ fn read_flash_info(mut transport: Transport) -> Result<()> {
     Ok(())
 }
 
+fn device_reset(mut transport: Transport) -> Result<()> {
+    // TODO allow users to choose their own reset method
+    transport.device_reset(ResetMethod::Reset)?;
```
Comment (sjoerdsimons) [inline rockusb/examples/rockusb.rs:35]: seems pretty straightforward to sort out that todo right away :)

File: rockusb/src/protocol.rs [hunk 4]
```diff
@@ -43,6 +43,16 @@ enum CommandCode {
...
 
+#[repr(u8)]
+#[derive(Debug, Eq, PartialEq, Clone, Copy, IntoPrimitive, TryFromPrimitive)]
+pub enum ResetMethod {
+	Reset = 0,
+	MSC,
+	PowerOff,
+	Maskrom,
+	Disconnect
+}
```
Comment (sjoerdsimons) [inline rockusb/src/protocol.rs:54]: Could you document what each of those  mean (to your undersatnding).

File: rockusb/examples/rockusb.rs (hunk 3 above)
Comment (obbardc) [reply rockusb/examples/rockusb.rs]: had some problems exposing the enum to clap, hopefully i did it right ;-)

File: rockusb/examples/rockusb.rs [hunk 5]
```diff
@@ -189,7 +195,8 @@ fn run_nbd(transport: Transport) -> Result<()> {
     Ok(())
 }
 
-#[derive(Debug, clap::Subcommand)]
+#[derive(Debug, clap::Parser)]
+// , clap::Subcommand
```
Comment (sjoerdsimons) [inline rockusb/examples/rockusb.rs:199]: leftover ?

**Pull request merged**